src/background/messaging/rpc-methods/stx-call-contract.ts

```diff
@@ -33,17 +33,20 @@ export async function rpcStxCallContract(
   port: chrome.runtime.Port
 ) {
   const { id: requestId, method, params } = message;
-  validateRequestParams({
+  const { status } = validateRequestParams({
     id: requestId,
     method,
     params,
     port,
     schema: stxCallContract.params,
   });
+  if (status === 'failure') return;
   const requestParams: RequestParams = [
     ['requestId', requestId],
     ['request', createUnsecuredToken(getMessageParamsToTransactionRequest(message.params))],
   ];
+  if (params.network) requestParams.push(['network', params.network]);
```

Comment:
I think this should be relocated to `getStxDefaultMessageParamsToTransactionRequest`.

Reply:
This is what I meant here, if it isn't included in the legacy request token, then it won't be used to gen/sign the tx:
https://github.com/leather-io/extension/blob/c431de791de4366cac5e68375292b074eb5d763b/src/app/common/rpc/use-rpc-sip30-broadcast-transaction.ts#L37

Essentially, these are unused in the wallet they way they are added here?

Reply:
but it changes the current network, so applies with the `?? network`

Reply:
Ok, so I didn't understand that about requests in general with network, so if that is passed it actually changes the current network. I thought it just set the network passed in the transaction to broadcast.